template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Extensions/IntegratedWebClientServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Identity.Service.IntegratedWebClient.Extensions
+{ 
+    public static class IntegratedWebClientServiceCollectionExtensions
+    {
+        public static IServiceCollection AddIntegratedWebClient(this IServiceCollection services)
+        {
+            services.AddIntegratedWebClient(_ => { });
+            services.TryAddEnumerable(ServiceDescriptor.Scoped<IConfigureOptions<IntegratedWebClientOptions>, IntegratedWebClientOptionsSetup>());
```

Comment:
I think it might be less weird if you just always bind to config, rather than binding only if no-one called Configure before Add.  They can always override anything they didn't want from config in the configure.